plutus-core/testlib/PlutusCore/Generators/Internal/Builtin.hs

```diff
@@ -71,6 +72,7 @@ genConstant tr
     | Just HRefl <- eqTypeRep tr (typeRep @()) = SomeGen $ pure ()
     | Just HRefl <- eqTypeRep tr (typeRep @Integer) = SomeGen genInteger
     | Just HRefl <- eqTypeRep tr (typeRep @Int) = SomeGen genInteger
+    | Just HRefl <- eqTypeRep tr (typeRep @Word8) = SomeGen genInteger
```

Comment:
I notice that `genInteger` only generates 'valid' ints or word8s, and we are not testing against a larger invalid / overflowed range

Reply:
I don't think it does?

```
>>> Gen.sample genInteger
270582141203701296
```

Reply:
What you gave me is smaller than `maxBound:: Int64` .

I meant that if an argument to a plutus builtinfunction is of type Int64 or Word8, we should feed that argument arbitrary large `Integer`s because that's our universe. 

With genInteger it seems that we generate a value between [minBound..maxBound]::Int64 and we cast that to Integer.

Reply:
Ah, you said "generated 'valid' ints or word8s" and I thought it would be easy to check the latter, which I did. Now that I look in the code we indeed only generate correct ints. @zliu41 could you create an issue for occasionally generating integers higher than `maxBound @Int64` and lower than `minBound @Int64`.

@bezirg thanks for spotting!